concent_api/common/decorators.py

```diff
@@ -85,3 +89,32 @@ def decorated(*args: Any, **kwargs: Any) -> None:
                 return func(*args, **kwargs)
 
         return decorated
+
+
+class retry_on_database_error(object):
+
+    def __init__(self, func: Callable) -> None:
+        self.func = func
+        self.max_number_of_retries = settings.MAXIMUM_NUMBER_OF_RETRIES  # pylint: disable=attribute-defined-outside-init
```

Comment:
There is no need to make it a class attribute. You never modify it, so you can use settings directly in `_validate_number_of_retries`.